src/Http/Controllers/SeatNotificationsController.php

```diff
@@ -52,42 +56,42 @@ public function index()
     public function postSubscribe(SubscribeRequest $request, SubscribeAction $action)
     {
 
-        if(empty($request->input('driver_id'))) {
+        if (! empty($request->input('group_id')) && empty($request->input('driver_id'))) {
             session([
                 'herpaderp.seatnotifications.subscribe.driver' => request()->input('driver'),
                 'herpaderp.seatnotifications.subscribe.notification' => request()->input('notification'),
+                'herpaderp.seatnotifications.subscribe.corporations_filter' => $request->input('corporations_filter'),
+                'herpaderp.seatnotifications.subscribe.characters_filter' => $request->input('characters_filter'),
             ]);
 
             return redirect()->route($request->input('notification')::getDriver($request->input('driver'))::getPrivateRegistrationRoute());
         }
 
         return $action->execute($request->all());
-
-        //dd($request, $request->input('client_id'), $request->input('driver'), $request->input('notification'));
-
     }
 
+    /**
+     * @param UnsubscribeAction $unsubscribe_action
+     * @return \Illuminate\Http\RedirectResponse|string
+     */
     public function getUnsubscribe(UnsubscribeAction $unsubscribe_action)
     {
 
-        if(request()->query('is_public'))
-            return 'is public';
-
-        $driver_id = request()->query('notification')::getDriver(request()->query('driver'))::getPrivateChannel();
-
         $data = [
             'driver' => request()->query('driver'),
-            'driver_id' => $driver_id,
+            'group_id' => request()->query('is_public') ? null : auth()->user()->group_id,
```

Comment:
the unique identifier is the combination of `driver_id` and `driver`. I am not much fan of the combiantion between group and a driver. Because group_id's might not be as stable as it needs to be (f.e. if you have an orphanaged group, it  gets deleted and next character registering receives that orphanaged group_id).

I think we should better think of creating:
`$driver_id = request()->query('notification')::getDriver(request()->query('driver'))::getPublicChannel();`

And in the driver we resolve the public channel subscribed to that notification?

Note: this is just an opinion. If you disagree we might need to alter the unique key on the migration file to accomendate.